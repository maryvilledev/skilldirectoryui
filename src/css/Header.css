/*
.Header {
  background-color: #555555;
  height: 80px;
  padding: 20px;
  padding-left: 10px;
  padding-top: 5px;
  color: white;
  text-align: left;
}

.Header-intro {
  font-size: large;
}

h1 {
  color: rgb(40, 185, 217);
  font-size: 23px;
  margin:0px;
  padding: 0px;
  width: inherit;
}
h3 {
  font-size: 12px;
  margin:0px;
  padding:0px;
}*/

@media (min-width: 768px)
.navbar-header {
  float: inherit;
}

@media (min-width: 768px)
.navbar-right {
    margin-right: 15px;
}
