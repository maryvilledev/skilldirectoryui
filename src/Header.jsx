import React from 'react';
import Grid from 'react-bootstrap/lib/Grid';
import { Link } from 'react-router';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.css';

import LoginLink from './LoginLink.jsx';


class Header extends React.Component {
  render() {
    return (
      <Grid fluid={true}>
        <Navbar bsStyle="default" fluid={true}>
          <Navbar.Header>
            <Navbar.Brand>
              <Link to="/">Skill Directory</Link>
            </Navbar.Brand>
          </Navbar.Header>
          {
          //Make sure all these paths are absolute (starting with a '/') so the
          //route will be handled properly
          }
          <Nav>
            <NavItem eventKey={1}>
              <Link to="/skills">Skills</Link>
            </NavItem>
            <NavItem eventKey={2}>
              <Link to="/team">Team</Link>
            </NavItem>
          </Nav>
          <Nav pullRight >
              <LoginLink />
          </Nav>
        </Navbar>
        {this.props.children}
      </Grid>
    );
  }
}
export default Header;
