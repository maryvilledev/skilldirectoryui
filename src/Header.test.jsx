import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Header from './Header';

describe('<Header />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });

  it('matches the stored snapshot', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.get(0)).toMatchSnapshot();
  });
});