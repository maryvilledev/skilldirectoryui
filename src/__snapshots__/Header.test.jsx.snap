exports[`<Header /> matches the stored snapshot 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={true}>
  <Uncontrolled(Navbar)
    bsStyle="default"
    fluid={true}>
    <NavbarHeader>
      <NavbarBrand>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/">
          Skill Directory
        </Link>
      </NavbarBrand>
    </NavbarHeader>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}>
      <NavItem
        active={false}
        disabled={false}
        eventKey={1}>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/skills">
          Skills
        </Link>
      </NavItem>
      <NavItem
        active={false}
        disabled={false}
        eventKey={2}>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/team">
          Team
        </Link>
      </NavItem>
    </Nav>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}>
      <NavItem
        active={false}
        disabled={false}>
        <LoginLink />
      </NavItem>
    </Nav>
  </Uncontrolled(Navbar)>
</Grid>
`;
